WcaOnRails/app/controllers/application_controller.rb

```diff
@@ -37,6 +55,15 @@ def doorkeeper_unauthorized_render_options(error: nil)
     devise_parameter_sanitizer.for(:account_update) << :name << :email
   end
 
+  # This method is called by devise after a successful login to know the redirect path
+  # We override it to do some action after signing in, but we want to use the original path
+  protected def after_sign_in_path_for(resource)
+    # When the user signs in, 'session[:locale]' is not cleared, so we need to update
+    # it with the user's preferred_locale
+    session[:locale] = current_user.preferred_locale || I18n.default_locale
```

Comment:
The idea was to prevent a locale which is not in the available locales to reach the session variable.
But yes I think simply calling `set_locale` is a good idea!